examples/service-clients/azure-client/external-controller/tests/index.ts

```diff
@@ -154,26 +146,34 @@ export async function createContainerAndRenderInElement(
 /**
  * For local testing we have two div's that we are rendering into independently.
  */
-async function setup() {
+async function setup(): Promise<void> {
+	// Since this is a single page Fluid application we are generating a new document id
+	// if one was not provided
+	const createNew = window.location.hash.length === 0;
+	if (createNew) {
+		window.location.hash = Date.now().toString();
+	}
+	const containerId = window.location.hash.substring(1);
+
 	const leftElement = document.getElementById("sbs-left") as HTMLDivElement;
 	if (leftElement === undefined) {
 		throw new Error("sbs-left does not exist");
 	}
-	await createContainerAndRenderInElement(leftElement, createNew);
+	await createContainerAndRenderInElement(containerId, leftElement, createNew);
 	const rightElement = document.getElementById("sbs-right") as HTMLDivElement;
 	if (rightElement === undefined) {
 		throw new Error("sbs-right does not exist");
 	}
 	// The second time we don't need to createNew because we know a Container exists.
-	await createContainerAndRenderInElement(rightElement, false);
+	await createContainerAndRenderInElement(containerId, rightElement, false);
 
 	// Setting "fluidStarted" is just for our test automation
 	// eslint-disable-next-line @typescript-eslint/dot-notation
 	window["fluidStarted"] = true;
 }
 
-setup().catch((e) => {
-	console.error(e);
+await setup().catch((error) => {
```

Comment:
Nit: since we're using top-level await here, should we just use a standard try-catch pattern?